Pipeline dependencies
==================================
This pipeline needs the following programs for running correctly.


=================================

- Bpipe:

http://code.google.com/p/bpipe/
Sadedin S, Pope B & Oshlack A, Bpipe: A Tool for Running and Managing Bioinformatics Pipelines, Bioinformatics

==================================

1) Quality Control

- HTQC:
http://sourceforge.net/projects/htqc/

License: GNU General Public License version 3.0 (GPLv3)

Xi Yang, Di Liu, Fei Liu, Jun Wu, Jing Zou, Xue Xiao, Fangqing Zhao and Baoli Zhu: HTQC: a fast quality control toolkit for Illumina sequencing data

BMC Bioinformatics 2013, 14:33

- Prinseq-lite and prinseq-graph: 

http://sourceforge.net/projects/prinseq/files/

License: GNU General Public License version 3.0 (GPLv3)

Schmieder R and Edwards R: Quality control and preprocessing of metagenomic datasets. Bioinformatics 2011, 27:863-864.

- fastx-toolkit

http://hannonlab.cshl.edu/fastx_toolkit/download.html

License: GNU General Public License version 3.0 (GPLv3) [ Except files under the "galaxy" sub-directory, that are distributed under the same license as Galaxy software ]

- Cutadapt:

http://code.google.com/p/cutadapt/

License: MIT License

MARTIN, M.. Cutadapt removes adapter sequences from high-throughput sequencing reads. EMBnet.journal, North America, 17, may. 2011.

- FASTQC:

http://www.bioinformatics.babraham.ac.uk/projects/fastqc/

License: GNU General Public License version 3.0 (GPLv3)

- Tally:

ftp://ftp.ebi.ac.uk/pub/contrib/enrightlab/kraken/reaper/src/reaper-latest/doc/tally.html

2) Clustering ( You can choose between different programs or install them all) 

- DnaClust:

http://sourceforge.net/projects/dnaclust/

License: GNU General Public License version 3.0 (GPLv3)

Mohammadreza Ghodsi, Bo Liu1 and Mihai Pop, DNACLUST: accurate and efficient clustering of phylogenetic marker genes BMC Bioinformatics 2011, 12:271

- Uclust: 

http://www.drive5.com/usearch/download.html

License: Specific (no open source)

Edgar,RC (2010) Search and clustering orders of magnitude faster than BLAST, Bioinformatics 26(19), 2460-2461.

- Cd-Hit:

http://weizhong-lab.ucsd.edu/cd-hit/download.php

License: GNU General Public License version 2 

Limin Fu, Beifang Niu, Zhengwei Zhu, Sitao Wu and Weizhong Li, CD-HIT: accelerated for clustering the next generation sequencing data. Bioinformatics, (2012), 28 (23): 3150-3152.

- MetaVelvet:

http://metavelvet.dna.bio.keio.ac.jp/

License: GNU General Public License version 3.0 (GPLv3)

Namiki T*, Hachiya T*, Tanaka H, Sakakibara Y. (2012) MetaVelvet : An extension of Velvet assembler to de novo metagenome assembly from short sequence reads, Nucleic Acids Res., in press. (Summplementary figures and tables) 

3) Annotation ( You can choose between FragGeneScan or Prodigal, or both)

- FragGeneScan: 

http://omics.informatics.indiana.edu/FragGeneScan/

License: GNU General Public License version 3.0 (GPLv3)

Mina Rho, Haixu Tang, and Yuzhen Ye. FragGeneScan: Predicting Genes in Short and Error-prone Reads. Nucl. Acids Res., 2010 

- Prodigal: 

http://prodigal.ornl.gov/

License: GNU General Public License version 3.0 (GPLv3)

Hyatt D, Chen GL, Locascio PF, Land ML, Larimer FW, Hauser LJ. Prodigal: prokaryotic gene recognition and translation initiation site identification. BMC Bioinformatics. 2010 Mar 8;11(1):119

